pilot/pkg/bootstrap/server.go

```diff
@@ -902,7 +902,12 @@ func (s *Server) initRegistryEventHandlers() {
 		if s.environment.GatewayAPIController != nil {
 			s.environment.GatewayAPIController.RegisterEventHandler(gvk.Namespace, func(config.Config, config.Config, model.Event) {
 				s.XDSServer.ConfigUpdate(&model.PushRequest{
-					Full:   true,
+					Full: true,
+					ConfigsUpdated: sets.Set[model.ConfigKey]{
+						model.ConfigKey{
+							Kind: kind.KubernetesGateway,
+						}: struct{}{},
```

Comment:
nit:   sets.New(model.ConfigKey{Kind: kind.KubernetesGateway, Name: ..., Namespace: ...}),

Reply:
We don't know the exact gateway name and namespace [here](https://github.com/istio/istio/blob/1.15.3/pilot/pkg/config/kube/gateway/controller.go#L338), instead we can just set the kind to make the gateway controller reconcile

Reply:
Then i doubt why ConfigsUpdated is useful here

Reply:
https://github.com/istio/istio/blob/34fe95198108bc603da5252361509fbbe731eabf/pilot/pkg/model/push_context.go#L1165-L1166

If `ConfigsUpdated` is empty, it will unnecessarily recompute a new context

Reply:
I am concerned the incomplete `ConfigsUpdated` will cause issue. e.g. xds cache clear depend on it

Reply:
Good point.. maybe we can change
```go
case kind.HTTPRoute, kind.TCPRoute, kind.GatewayClass, kind.KubernetesGateway, kind.TLSRoute, kind.ReferenceGrant:
	gatewayAPIChanged = true
	virtualServicesChanged = true
	gatewayChanged = true
```
to
```go
case ..., kind.Namespace:
	gatewayAPIChanged = true
	virtualServicesChanged = true
	gatewayChanged = true
```
and add a comment for it? Hope it's not that confusing, WDYT